### DISCUSSION TITLE
生成的pdf文件可以具备多级目录吗（不止3级）

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#124] [Q&A]
生成的pdf文件可以具备多级目录吗（不止3级），文章中的目录可以调到多级不

**Comment by sikouhjw** [comment]
可以。但请你看看规范是怎么写的。

**Reply by earthmoon** [reply]
文内目录三级还行，但是生成的pdf导航栏可以格外设置成多级吗？这样的话使用pdf会舒服一点。还有pdf的导航栏部分是不是不支持上下标这种？

**Reply by tanukihee** [reply]
（看得我很好奇，我校我院在格式要求上确实松，但有这么松吗）

**Comment by tanukihee** [comment]
可以，请自由更改

**Reply by earthmoon** [reply]
文内目录和pdf导航栏的显示的级别数目不一样也可以吗

**Reply by tanukihee** [reply]
可以，这个应该是由 hyperref 控制的

**Reply by earthmoon** [reply]
> 以，这个应该是由 hyperref 控制的

好的，谢谢。我先留意，等最后统一弄一下，现在水平太菜了。

**Reply by BenjaminHb** [reply]
可以通过修改这里调整目录显示层级。

https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/whu-thesis.cls#L300

**Reply by earthmoon** [reply]
> 可以通过修改这里调整目录显示层级。
> 
> https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/whu-thesis.cls#L300

谢谢！我想问下，

我如果把`secnumdepth    = 5,  % 标题层级数`改为`secnumdepth    = 8,  % 标题层级数`， 我是不是可以用`\subsubsubsubsection`命令了？

生成的pdf文档目录只保留3级，因为学校要求3级，但是pdf导航栏想多级，因为pdf导航多级更方便。

这两个都可以么

**Reply by xkwxdyy** [reply]
不存在 `\subsubsubsubsection`

**Reply by earthmoon** [reply]
> 不存在 `\subsubsubsubsection`

`secnumdepth = 5` 这个的意思是什么呢

**Reply by xkwxdyy** [reply]
![image](https://user-images.githubusercontent.com/87134063/163588694-ec1ae125-4b19-45ec-ace4-643e3c7b5004.png)

**Reply by tanukihee** [reply]
你为什么不试试神奇的百度呢

STFW RTFM

**Reply by earthmoon** [reply]
> ![image](https://user-images.githubusercontent.com/87134063/163588694-ec1ae125-4b19-45ec-ace4-643e3c7b5004.png)

那这个5岂不是最大了。。。

**Reply by tanukihee** [reply]
不建议没有三个月基础的初学者使用 LaTeX 进行工程级别文章写作

**Reply by xkwxdyy** [reply]
非常建议你起码完整读一遍 lshort-zh-cn 再使用

**Reply by earthmoon** [reply]
> 非常建议你起码完整读一遍 lshort-zh-cn 再使用

好的，我有时间再读一遍。

**Reply by earthmoon** [reply]
> 不建议没有三个月基础的初学者使用 LaTeX 进行工程级别文

不是写大论文，谁刻意用这个。本来排版就是编辑的事，大论文没有编辑来处理，只有自己干。就差cls不懂，平常使用还好吧。至少有两个宏包的作者会回复邮件。虽然他们觉得我的问题很简单，但是还没有要我不要继续，也没有说我看的太少了。其中一个宏包作者告诉我，看什么都没用，就是要边做边学，latex不是两本书可以解决的。他就是这么和我说的。

**Reply by tanukihee** [reply]
别等有时间了，现在就读

在没有一定基础的前提下纠结与写作不甚相关的问题完全是浪费时间

我没有不让你用，我认为有两件事是非常重要的。第一件事是：凡事都有先易后难，在纠结高级内容（比如字体样式、pdf 书签、某个图片的具体位置等）之前，请先把基本的东西（文章内容、数学公式、LaTeX 基本使用方法等）解决，至少不会问出一些非常非常基础的问题；而第二件事是：在使用任何工具之前，先读说明书，在说明书读不懂的情况下，再查询互联网的已有内容，最后保证你的问题在上两渠道都无法得到解决的情况下，最后询问他人。虽然这个比较难以做到，但至少可以避免问出许多让人感觉「完全没有动脑」的问题来

你的许多问题，在文档中都有相应的说明，互联网上也有大把的答案，当然，也欢迎你在后续的使用过程中，提出[高质量的问题](https://github.com/tvvocold/How-To-Ask-Questions-The-Smart-Way)

**Reply by tanukihee** [reply]
> 看什么都没用

但至少不看是完全没用的

> 就是要边做边学

在边做边学前面应该加上「边看」，边看边做边学

> latex 不是两本书可以解决的。

的确如此，但你的这些基础问题是可以的。「不是两本书可以解决」的问题，大多数的使用者都碰不到，即使碰到了，三四本乃至一二十本书总能解决的，而一本书不看是完全解决不了任何问题的

**Reply by sikouhjw** [reply]
> 至少有两个宏包的作者会回复邮件。虽然他们觉得我的问题很简单，但是还没有要我不要继续，也没有说我看的太少了。其中一个宏包作者告诉我，看什么都没用，就是要边做边学，latex不是两本书可以解决的。他就是这么和我说的。

不要把宏包作者的耐心花在这种地方，珍惜给你解答的人、机会，耐心耗完就没了的。

**Reply by earthmoon** [reply]
> > 至少有两个宏包的作者会回复邮件。虽然他们觉得我的问题很简单，但是还没有要我不要继续，也没有说我看的太少了。其中一个宏包作者告诉我，看什么都没用，就是要边做边学，latex不是两本书可以解决的。他就是这么和我说的。
> 
> 不要把宏包作者的耐心花在这种地方，珍惜给你解答的人、机会，耐心耗完就没了的。

每个人的态度不一样，所以别人才说国内没有好论坛。有一位美国顶级教授指导我做实验，3个月，我一直都不好意思，都没见过。从那之后，任何人问我问题，我都会像那位教授一样，最后来一句：你还有哪里不懂。我之前也觉得别人的问题太傻什么的，从那之后，无论问题还是人，一律平等。

**Reply by sikouhjw** [reply]
学术问题和开源问题差别很大，开源是纯粹的公益行为，而公益的回报是边际效应递减的。

国外的[提问的智慧](https://github.com/tvvocold/How-To-Ask-Questions-The-Smart-Way)正是表达了开源问题的特殊性，能读书解决的却没有读，就跟老师讲过的内容还没听是一个道理。

国外的论坛也不见得友好，[tex.se](https://tex.stackexchange.com/) 对于重复的问题、没有 MWE 的问题是很不客气的，如果持续的不提供 MWE，是会锁定问题的。

只是人家平台基数大，热心人多，一片热心坚持到最后的总是少数，双方互利共赢、互相方便才是最优解，所以多读手册、多搜，起码你要让别人知道你搜过了（例如：[https://github.com/sikouhjw/gdutthesis/discussions/66](https://github.com/sikouhjw/gdutthesis/discussions/66)，自己尝试过的再提的问题，只会让作者更有动力去回答）。

**Comment by earthmoon** [comment]
只要文内目录保持三级，就是合规的。pdf导航栏多级是方便阅读而已。

**Comment by earthmoon** [comment]
也许吧。个人觉得最重要的是要找到对应的宏包，有些功能没有宏包可以直接实现，就很难了。目前表格这块没有什么问题了，表格宏包作者，他先帮我解决了我要的排版，然后和我详细说代码。图片处理的好的宏包还没找到，所以图片就是默认的排版。公式正在琢磨。其他无伤大雅，能实现就实现，实现不了也没事的。你们的模板在这，五脏俱全。大论文可以应付。
________________________________

**Reply by xkwxdyy** [reply]
表格：`tabularray`
图片：一些常用的图片宏包 lshort-zh-cn 都有介绍

**Reply by earthmoon** [reply]
> 表格：`tabularray`
> 
> 图片：一些常用的图片宏包 lshort-zh-cn 都有介绍

表格这个包基本够用了。图片目前来看没有特别好的包。比如图片的注释，目前借助minipage。图片的文字环绕包wrapfig太老了，有缺陷。

**Reply by xkwxdyy** [reply]
为什么非要绕排呢

**Reply by earthmoon** [reply]
> 基本够用了。图片目前来看没有特别好的包。比如图

因为有些图片小，然后想着都用latex了，不如精益求精，就当作一本书的排版来。等我以后有大量时间，我会把latex学透，然后基于这个模板，把我认为不足的都改进，有违规定的就提供接口给使用者。这只是一个想法，以后也许坚持不下去。

**Reply by sikouhjw** [reply]
> 因为有些图片小，然后想着都用latex了，不如精益求精，就当作一本书的排版来。等我以后有大量时间，我会把latex学透，然后基于这个模板，把我认为不足的都改进，有违规定的就提供接口给使用者。这只是一个想法，以后也许坚持不下去。

有些需求，跟学位论文一点关系都没有，实现进去只会更加臃肿和难以维护。

**Reply by earthmoon** [reply]
> > 因为有些图片小，然后想着都用latex了，不如精益求精，就当作一本书的排版来。等我以后有大量时间，我会把latex学透，然后基于这个模板，把我认为不足的都改进，有违规定的就提供接口给使用者。这只是一个想法，以后也许坚持不下去。
> 
> 有些需求，跟学位论文一点关系都没有，实现进去只会更加臃肿和难以维护。

也有道理。

**Reply by xkwxdyy** [reply]
绕排有一个新宏包 `wrapstuff` 可以参考

**Reply by earthmoon** [reply]
> 绕排有一个新宏包 `wrapstuff` 可以参考

好的，谢谢您！

**Comment by earthmoon** [comment]
我在tex.se问了问题，没有MWE。然后他发了链接教我怎么做。在主题浏览下，一直回复我。直到最后我和他说，我做出的MWE反应不出我的问题，我的问题是我们学校模板才有的。然后他就要我把模板发给他。确实后来那个问题因为没有MWE而被关闭了。每天早上起来第一件事就是看SE是我的常态了，无论学术还是技术，那里都有坚持者。我很喜欢那个论坛，国外好论坛很多。
________________________________
